app/src/org/commcare/preferences/FilePreferenceDialogFragmentCompat.java

```diff
@@ -47,7 +52,12 @@ public void onActivityResult(int requestCode, int resultCode, Intent intent) {
         if (requestCode == REQUEST_FILE) {
             if (resultCode == RESULT_OK && intent != null) {
                 Uri uri = intent.getData();
-                String filePath = UriToFilePath.getPathFromUri(CommCareApplication.instance(), uri);
+                String filePath;
+                try {
+                    filePath = UriToFilePath.getPathFromUri(CommCareApplication.instance(), uri);
+                } catch (UriToFilePath.NoDataColumnForUriException e) {
+                    filePath = getNewPathFromUri(uri);
+                }
                 validateFile(filePath);
```

Comment:
if `getNewPathFromUri` above returns null we'll call `Toast.makeText()` twice in quick succession here https://github.com/dimagi/commcare-android/blob/8b4982199f3d9b45f795996ca783c249f554e956/app/src/org/commcare/preferences/FilePreferenceDialogFragmentCompat.java#L59-L59 and then here https://github.com/dimagi/commcare-android/blob/8b4982199f3d9b45f795996ca783c249f554e956/app/src/org/commcare/preferences/FilePreferenceDialogFragmentCompat.java#L119-L119

Might be better to throw an exception from `getNewPathFromUri` and handle that rather than returning null